api/src/main/java/org/pmiops/workbench/api/RuntimeController.java

```diff
@@ -173,10 +176,69 @@ private DbWorkspace lookupWorkspace(String workspaceNamespace) throws NotFoundEx
         workspaceNamespace, firecloudWorkspaceName, WorkspaceAccessLevel.WRITER);
     workspaceService.validateActiveBilling(workspaceNamespace, firecloudWorkspaceName);
 
-    return ResponseEntity.ok(
-        leonardoMapper.toApiRuntime(
-            leonardoNotebooksClient.getRuntime(
-                workspaceNamespace, userProvider.get().getRuntimeName())));
+    try {
+      return ResponseEntity.ok(
+          leonardoMapper.toApiRuntime(
+              leonardoNotebooksClient.getRuntime(
+                  workspaceNamespace, userProvider.get().getRuntimeName())));
+    } catch (NotFoundException e) {
+      if (!workbenchConfigProvider.get().featureFlags.enableCustomRuntimes) {
+        throw e;
+      }
+
+      try {
+        return ResponseEntity.ok(getOverrideFromListRuntimes(workspaceNamespace));
+      } catch (RuntimeException e2) {
+        throw e;
+      }
+    }
+  }
+
+  private Runtime getOverrideFromListRuntimes(String workspaceNamespace) {
+    Optional<LeonardoListRuntimeResponse> mostRecentRuntimeMaybe =
+        leonardoNotebooksClient.listRuntimesByProject(workspaceNamespace, true).stream()
+            .sorted(
+                (a, b) -> {
+                  String a_createdDate, b_createdDate;
+                  if (a.getAuditInfo() == null || a.getAuditInfo().getCreatedDate() == null) {
+                    a_createdDate = "";
+                  } else {
+                    a_createdDate = a.getAuditInfo().getCreatedDate();
+                  }
+
+                  if (b.getAuditInfo() == null || b.getAuditInfo().getCreatedDate() == null) {
+                    b_createdDate = "";
```

Comment:
Wait - these are strings? If so, I don't think the comparison below is going to work as expected. Probably need a date / instant parsing step in here.

Reply:
I think it works as long as we receive the timestamps in a consistent timezone. 

"easily comparable and sortable with a trivial string comparison"
https://fits.gsfc.nasa.gov/iso-time.html

Reply:
going to merge since I have approval and tests are passing but I can add something later if this is still a concern

Reply:
Oh wow, wasn't aware of that property. That's quite useful, thanks.